public/app/features/browse-dashboards/state/hooks.ts

```diff
@@ -83,3 +95,38 @@ function createFlatTree(
 
   return items.flatMap((item) => mapItem(item, folderUID, level));
 }
+
+/**
+ * Returns a DashboardTreeSelection but removes any selected folder's children.
+ * This is useful when making backend requests to move or delete items.
+ * In this case, we only need to move/delete the parent folder and it will cascade to the children.
+ * @param selectedItems Overall selection state
+ * @param childrenByParentUID Arrays of children keyed by their parent UID
+ */
+function getSelectedItemsForBackend(
```

Comment:
Took me a bit to properly grok how this works, but this is a really smart and neat solution. Well done 👍🏻